### PR TITLE
processing data in models

### PR DESCRIPTION
**Pull request by ANTOINE1128**
# In this PR I have followed these project requirements:
1- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
2- Set up associations between models.
Remember that author_id column in posts table should be the foreign_key for the users table.
3 - Create custom methods

### CODE REVIEW

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @ANTOINE1128 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] The associations between models has been setup correctly👍🏼.
- [x] User model has method that returns 3 most recent posts👍🏼.
- [x] Post model has a method that updates the posts counter for a user👍🏼.


## Required Changes ♻️

- [ ] Nice job so far👍🏼. currently, you haven't implemented a method that updates the comment counter in the comment model, for a post as required in this milestone😉. kindly do well to implement that👍🏼.
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
   belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  def update_comment_counter
+    post.update(comment_counter: post.comments.count)
+  end
```
Comment (codepantha) [inline app/models/comment.rb:7]: - [ ] [OPTIONAL] Please consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comment_counter` method every time a new comment gets saved to database, then make the `update_comment_counter` method private.

File: app/models/post.rb [hunk 2]
```diff
@@ -2,4 +2,15 @@ class Post < ApplicationRecord
...
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  def update_likes_counter
+    update(likes_counter: likes.count)
+  end
```
Comment (codepantha) [inline app/models/post.rb:15]: - [ ] [OPTIONAL] Please consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `#update_post_counter` method every time a new post gets saved to database, then make the `update_post_counter` method private.
**NB** Please do same for the `update_likes_counter` method 👍

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @Ibizugbe 👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Set up association between models 📌 
✔️ No linter errors 🟢 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**